km62/Kozyriev_Anton/laboratory1.sql

```diff
@@ -1,2 +1,141 @@
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина купляє певну марку телефону.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE TABLE Customer
+(
+  customer_name VARCHAR(30) NOT NULL
+);
+
+ALTER TABLE Customer ADD CONSTRAINT customer_name PRIMARY KEY;
+
+CREATE TABLE Phone
+(
+  phone_model VARCHAR(30) NOT NULL
+);
+
+ALTER TABLE Phone ADD CONSTRAINT phone_model PRIMARY KEY;
+
+CREATE TABLE Phone_brand
+(
+  customer_name VARCHAR(30) NOT NULL,
+  phone_model VARCHAR(30) NOT NULL,
+  phone_brand VARCHAR(30) NOT NULL
+);
+
+ALTER TABLE Phone ADD CONSTRAINT phone_model PRIMARY KEY;
+ALTER TABLE Phone ADD CONSTRAINT customer_name_fk FOREIGN KEY (Customer) references customer_name;
+ALTER TABLE Phone ADD CONSTRAINT phone_model_fk FOREIGN KEY (Phone) references phone_model;
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+GRANT CREATE ANY TABLE TO kozyrev
+GRANT UPDATE ANY TABLE TO kozyrev
+GRANT SELECT ANY TABLE TO kozyrev;
+
+/*---------------------------------------------------------------------------
+3.a. 
+Яка ціна найдорожчого товару, що нікому не було продано?
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+
+SELECT CUST_NAME AS "Customer_name"
+FROM 
+(
+  SELECT CUST_NAME
+  FROM Customers, OrderItems, Orders
+  WHERE Orderitems.ITEM_PRICE = MAX(ITEM_PRICE)
+  AND Customers.CUST_ID = Oreders.CUST_ID
+  AND Oreders.ORDER_NUM = Oreders.ORDER_NUM
+);
+
+
+
+
+/*---------------------------------------------------------------------------
+3.b. 
+Як звуть покупця, у якого у замовленні найдорожчий товар – поле назвати Customer_name.
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+SELECT CUST_NAME AS "Customer_name"
+FROM 
+(
+  SELECT CUST_NAME
+  FROM Customers, OrderItems, Orders
+  WHERE Orderitems.ITEM_PRICE = MAX(ITEM_PRICE)
+  AND Customers.CUST_ID = Oreders.CUST_ID
+  AND Oreders.ORDER_NUM = Oreders.ORDER_NUM
+);
+
+
+/*---------------------------------------------------------------------------
+c. 
+Вивести ім’я та країну постачальника, як єдине поле vendor_name, для тих остачальників, що не мають товару.
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
```

Comment:
SELECT concat(VENDORS.vend_name, VENDORS.VEND_COUNTRY)
FROM VENDORS WHERE NOT EXISTS(SELECT * FROM PRODUCTS WHERE Products.vend_id = Vendors.vend_id);

Reply:
Fixed!